sui_core/src/full_node/full_node_state.rs

```diff
@@ -49,9 +52,12 @@ const MAX_TX_RANGE_SIZE: u64 = 4096;
 
 pub struct FullNodeState {
     pub store: Arc<ReplicaStore>,
+    pub indexes: Arc<IndexStore>,
     pub committee: SuiCommittee,
     pub next_tx_seq_number: AtomicU64,
 
+    tx_notify: Notify,
```

Comment:
Whats the purpose of this?

Reply:
See below - we need a way for the tests to wait until a transaction has synced to the node, otherwise we need to use hard-coded delays which will be flaky.

Reply:
So this is only used for testing purposes? seems like the more appropriate way to do this would be to poll the node via the rpc endpoint for the txn digest. but i may be missing something

Reply:
Well, it might not matter much in practice, but there's no easy way to do that without a race, so we could still have a flaky test - we might end up testing the indexes after the ReplicaStore is updated but before IndexStore has been updated. Also, using polling would turn some test breakages into timeouts instead of failures which will be annoying to debug. (The existing code can timeout too of course, but there are fewer situations where it can).

The right way to do this may be to eventually subscribe to the subscription API that we'll need for events anyway. How about I add a TODO to do that and leave the current testing code in?